synapse/util/wheel_timer.py

```diff
@@ -84,7 +87,7 @@ def fetch(self, now: int) -> List[T]:
         """
         now_key = int(now / self.bucket_size)
 
-        ret = []
+        ret: List[T] = []
         while self.entries and self.entries[0].end_key <= now_key:
             ret.extend(self.entries.pop(0).queue)
```

Comment:
I was going to say "where do we remove elements from `_Entry.queue`. Looks like we consume each queue/bucket as a whole and them drop them for the GC to reclaim.